### DISCUSSION TITLE
Semantic segmentation-aireal

### DISCUSSION THREAD

**Discussion by itaim18** [ultralytics/ultralytics#18103] [Q&A]
I have frames and images of drones.
I want to train a model for semantic segmentation of it including: roads, roundabouts, crosswalks, junctions etc.
What tools and methods should I use first this task.
Would hate to go over all frames one by one etc.
Thinking of SAM 2 as a helper but don’t see how exactly…
Any ideas?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @itaim18, thank you for reaching out and sharing your use case for semantic segmentation with aerial drone images 🚀! This sounds like an exciting project, and we’re here to help guide you in the right direction.

We recommend first checking out the [Ultralytics Docs](https://docs.ultralytics.com) for resources on the tasks you wish to accomplish. Semantic segmentation tasks may also benefit from reviewing the [model training tips](https://docs.ultralytics.com/guides/model-training-tips/) and popular segmentation techniques for optimal results.

If this is a ❓ general question about workflows or methods, could you please provide more details about your dataset or tools you've considered (like SAM)? This will help provide additional clarity for recommendations. If you're considering using SAM, you might want to look into its applications for data labeling and model initialization, as these can significantly reduce your manual efforts in preparing segmentation datasets.

If this is a 🐛 technical issue or bug report related to the tools or methods, we kindly ask you to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/), along with error logs or screenshots, to help us identify and debug the issue.

---

### Key Resources to Explore 🔍

- [Python Usage Guide](https://docs.ultralytics.com/usage/python/) – Learn about integrating Python for task automation.
- [CLI Usage Guide](https://docs.ultralytics.com/usage/cli/) – Get started with command-line interface workflows.
- [Custom Training Guide](https://docs.ultralytics.com/guides/model-training-tips/) – Improve your model training with segmentation-specific tips.

### Explore the Community 🎧

Join and engage with the vibrant Ultralytics community for support or tips:
- [Discord](https://discord.com/invite/ultralytics) for real-time discussions.
- [Discourse](https://community.ultralytics.com) for in-depth conversations.
- [Subreddit](https://reddit.com/r/Ultralytics) for knowledge sharing and project inspiration.

---

### Upgrade Ultralytics 🔄

Make sure you’re running the latest version of the `ultralytics` package for new features and bug fixes. Upgrade it in your environment with:

```bash
pip install -U ultralytics
```

Verify that you are using a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).

---

### Verified Environments 🖥️

YOLO is tested and verified to work seamlessly in the following environments:
- **Notebooks** with a GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Cloud Platforms: [GCP](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) and [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).
- [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

---

### CI Status ✅

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>  
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing, ensuring the reliability of YOLO in supported environments.

---

🤖 This is an automated response, and an Ultralytics engineer will assist you further soon! 🎉

**Reply by itaim18** [reply]
This is a general question. Unfortunately, I don't believe the visdrone dataset would help much in achieving training my model.

**Reply by glenn-jocher** [reply]
@itaim18 thank you for your feedback. If the VisDrone dataset isn't suitable, I recommend exploring datasets like DeepGlobe or OpenStreetMap for aerial imagery segmentation tasks, or you can leverage SAM2 for efficient annotation of your custom dataset. Let us know if you need further assistance with this!

**Reply by itaim18** [reply]
Hey, @glenn-jocher , Let's say I was able to segment abt 150 images into roboflow, how should i train my model from here?
would you recommend one of their notebooks to do it? was looking at this: https://github.com/roboflow/notebooks/blob/main/notebooks/fine-tune-sam-2.1.ipynb but not sure if this is the right approach...

**Reply by glenn-jocher** [reply]
@itaim18 you can train your model using Roboflow's tools or export your dataset in YOLO format and train it directly with the Ultralytics framework. Both approaches are effective, but if you're focused on YOLO models, training within the Ultralytics ecosystem ensures alignment with the model's architecture and optimization.

**Reply by Y-T-G** [reply]
@itaim18 Ultralytics doesn't support semantic segmentation. It trains an instance segmentation model. And you need lots of data. That's how deep learning works.

**Comment by glenn-jocher** [comment]
@itaim18 to train a semantic segmentation model for aerial imagery, you can leverage Ultralytics tools for auto-annotation. Use the `auto_annotate` function in our framework to efficiently label your dataset by combining a YOLO detection model (e.g., YOLO11) with a segmentation model like SAM or SAM 2. This will save you from manually labeling each frame. Refer to the [SAM 2 documentation](https://docs.ultralytics.com/models/sam-2/) for integration and usage details.

**Comment by itaim18** [comment]
Hey :)
I actually figured some stuff in this area, probably gonna leverage the project’s abilities in the few days and look more into it
Thank you very much!!

**Reply by glenn-jocher** [reply]
@itaim18 you're welcome! If you have any further questions while exploring the project, feel free to reach out. Best of luck with your work!
